### PR TITLE
Fix ARM trailing data directive size

### LINKED ISSUE #382: ARM/Thumb: Final data directive size is incorrect
On the latest version (3.7.3), the final data directive of a function is sometimes the wrong directive. 

Here's an example of it in action, causing the function to be unable to show 100%:

Image

For reference, the original input assembly is `.4byte 0x0000FFFF`.

Here is a relevant function of it happening from decomp.me:

[_KLEOD__sub_08014184.zip](https://github.com/user-attachments/files/30153329/_KLEOD__sub_08014184.zip)

### PR DESCRIPTION
**Pull request by floze-the-genius**
## Summary
- initialize ARM mapping symbols before inferring zero-sized function spans
- preserve terminal word-sized `$d` data directives while retaining trailing halfword padding removal
- add the issue #382 object fixture and regression test

## Testing
- `cargo test`

Closes #382

### CODE REVIEW

File: objdiff-core/src/arch/arm.rs [hunk 1]
```diff
@@ -474,11 +478,21 @@ impl Arch for ArchArm {
...
+                    .get(&section_idx)
+                    .and_then(|mappings| {
+                        mappings
+                            .iter()
+                            .rfind(|mapping| mapping.address as u64 <= next_address - 2)
+                    })
+                    .is_some_and(|mapping| {
+                        mapping.mapping == unarm::ParseMode::Data
+                            && next_address - mapping.address as u64 >= 4
+                    })
```
[the remark this answers is not in the document]
Comment (floze-the-genius) [reply objdiff-core/src/arch/arm.rs]: Fixed in e2b98eb. The ARM data guard now preserves only complete 4-byte data words, so a 4n+2 `$d` range trims the trailing zero halfword and stops at 4n. I also kept relocation overlap protection while trimming and added boundary coverage for 2/4/6/8-byte ranges. ARM integration tests (including the relocation snapshot) and the focused unit test pass; fmt and clippy pass as well (with only the repository’s removed generated-code lint name explicitly allowed).

File: objdiff-core/src/arch/arm.rs [hunk 2]
```diff
@@ -474,20 +478,24 @@ impl Arch for ArchArm {
...
+                        mappings.iter().rfind(|mapping| mapping.address as u64 <= next_address - 2)
+                    })
+                    .is_some_and(|mapping| mapping.ends_with_complete_data_words(next_address))
+            })
             && let Some(data) = section.data_range(next_address - 2, 2)
             && data == [0u8; 2]
+            && !section.relocation_at(next_address.saturating_sub(4), 4).is_some_and(|relocation| {
+                relocation.address + self.data_reloc_size(relocation.flags) as u64
+                    > next_address - 2
+            })
```
[the remark this answers is not in the document]
Comment (floze-the-genius) [reply objdiff-core/src/arch/arm.rs]: Fixed in 2c6124b. `Section::relocations_at` now exposes every relocation in the queried range, and the ARM trailing-halfword guard checks all of them for overlap instead of stopping at the first. Added a regression test with adjacent `R_ARM_ABS16` relocations at `end - 4` and `end - 2`; the focused unit test and all 8 ARM integration tests pass.

File: objdiff-core/src/arch/arm.rs [hunk 3]
```diff
@@ -500,6 +518,11 @@ struct DisasmMode {
 }
 
 impl DisasmMode {
+    fn ends_with_complete_data_words(self, end_address: u64) -> bool {
+        let data_size = end_address.saturating_sub(self.address as u64);
+        self.mapping == unarm::ParseMode::Data && data_size >= 4 && data_size.is_multiple_of(4)
```
[the remark this answers is not in the document]
Comment (floze-the-genius) [reply objdiff-core/src/arch/arm.rs]: Fixed in 6144e2e. `ends_with_complete_data_words` now removes the bytes needed to reach the next 4-byte boundary before checking the complete-word portion, so a `$d` mapping at `2 mod 4` correctly treats its leading halfword separately. The unit test now covers both aligned and `0x1002` mappings, including complete and incomplete trailing words; all 8 ARM integration tests pass.

**Comment by floze-the-genius** [comment]
@LagoLunatic Thanks for supplying the corrected object fixture on #382. The three edge cases raised during automated review are now covered and all threads are resolved: 4n+2 padding, overlapping relocations in the final word, and data mappings starting at 2 mod 4. Current head `64cf492` is mergeable; 19 checks pass (4 release-only jobs are skipped). Could you review whether the ARM size-inference behavior now matches the intended mapping-symbol semantics? I’m happy to rebase onto current main if you prefer that before review.
